### PR TITLE
Jssidebar macro updates to add learning area links

### PR DESCRIPTION
**Pull request by chrisdavidmills**
In this branch, I've updated the jssidebar macro to add in sensible learning area links, to make the complete beginner's material easier to find.

### CODE REVIEW

**Comment by Elchi3** [comment]
This contains commits from your other PR and now conflicts :(

**Comment by chrisdavidmills** [comment]
I should have resolved this conflict now.

File: macros/JsSidebar.ejs [hunk 1]
```diff
@@ -28,10 +29,11 @@ var text = mdn.localString({
     'Guide_OOP': 'Details of the object model',
     'Guide_Iterators_Generators': 'Iterators and generators',
     'Guide_Meta': 'Meta programming',
-    'Introductory': 'Introductory',
+    'Complete_beginners': 'Complete beginners',
     'Basics': 'JavaScript basics',
-    'Overview': 'JavaScript technologies overview',
-    'OO_JavaScript': 'Introduction to Object Oriented JavaScript',
+    'First_steps': 'JavaScript first steps',
+    'Building_blocks': 'JavaScript Building blocks',
```
Comment (Elchi3) [inline macros/JsSidebar.ejs]: JavaScript building blocks

**Comment by Elchi3** [comment]
This puts https://developer.mozilla.org/en-US/docs/Web/JavaScript/JavaScript_technologies_overview as the first article into the navigation. I like the article, but do we think this should be the first thing there?

File: macros/JsSidebar.ejs (hunk 1 above)
Comment (chrisdavidmills) [inline macros/JsSidebar.ejs]: Looks ok to me ;-)

**Comment by chrisdavidmills** [comment]
I think we agreed to put the JavaScript technologies overview into it's own top level item, but we never agreed on where ;-) I am happy with whatever you think is best. An overview suggests that it should be early on, but really it is the kind of thing that more experienced devs would be interested in. What about at the end of the References section? It is more of a reference than a tutorial.

File: macros/JsSidebar.ejs (hunk 1 above)
Comment (Elchi3) [inline macros/JsSidebar.ejs]: I meant the lower case b, which I think is correct according to our style guide.

**Comment by Elchi3** [comment]
I agree an overview should be early on, but I think this one is quite technical as you said. Let's put it in the "Misc" tree as the first item?

**Comment by chrisdavidmills** [comment]
Ah, "Building Blocks" was a typo - I've fixed it now. I also agree with you about the placing over the overview article - I've moved it.

**Pull request merged**